pkg/cpio/newc.go

```diff
@@ -148,17 +146,19 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 	}
 
 	// Some files do not have any content.
-	if f.ReadCloser == nil {
+	if f.ReaderAt == nil {
 		return nil
 	}
 
 	// Write file contents.
-	m, err := io.Copy(w, f)
+	m, err := io.Copy(w, io.NewSectionReader(f, 0, math.MaxInt64))
```

Comment:
Since `io.NewSectionReader(f, 0, math.MaxInt64)` is used multiple times, perhaps replace with something like `uio.ForkReader(f)`.

Reply:
Done.